File: cpp/creational/prototype/nishith/Prototype/Prototype/RealPlayer.cpp
#include "RealPlayer.h"
#include <iostream>

RealPlayer::RealPlayer(const std::string& name, int health, int damage)
   : Player(name, health, damage)
{

}

void RealPlayer::print() const
{
   std::cout << "RealPlayer: " << name_ << ", Health: " << health_
      << ", Damage Output: " << damage_ << std::endl;
}

Player* RealPlayer::clone() const
{
   return new RealPlayer(*this);
}

File: java/creational/prototype/praful/RealPlayer.java
public class RealPlayer extends Player {
    private String weapon;

    public RealPlayer(String name, String weapon, int health, int level) {
        super(name, health, level); // Call the constructor of Player with level
        this.weapon = weapon; // Initialize the weapon
    }

    public RealPlayer(RealPlayer realPlayer) {
        super(realPlayer.getName(), realPlayer.getHealth(), realPlayer.getLevel()); // Copy constructor
        this.weapon = realPlayer.weapon; // Copy the weapon
    }

    @Override
    public Player clone() {
        return new RealPlayer(this); // Create a new RealPlayer using the copy constructor
    }

    @Override
    public void print() {
        System.out.println("RealPlayer Name: " + getName());
        System.out.println("Health: " + getHealth());
        System.out.println("Level: " + getLevel());
        System.out.println("Weapon: " + weapon);
    }
}
